server/src/main/java/org/opensearch/common/settings/WriteableSetting.java

```diff
@@ -85,7 +86,12 @@ public WriteableSetting(StreamInput in) throws IOException {
         if (hasFallback) {
             fallback = new WriteableSetting(in);
         }
-        // We are using known types so don't need the parser
+        // Read the parser
+        WriteableSetting parser = null;
+        boolean isParserWriteable = in.readBoolean();
+        if (isParserWriteable) {
+            parser = new WriteableSetting(in);
+        }
```

Comment:
After you read the (`Writeable`) parser object you'll need to pass it to `createSetting()` on line 99 below.